### PR TITLE
implemented logger in d2gamescreen

### PR DESCRIPTION
**Pull request by gucio321**
Hi there,
no more import "log" in module d2gamescreen

### CODE REVIEW

File: d2app/app.go [hunk 1]
```diff
@@ -829,7 +829,7 @@ func (a *App) quitGame() {
 }
 
 func (a *App) enterGuiPlayground() {
-	a.screen.SetNextScreen(d2gamescreen.CreateGuiTestMain(a.renderer, a.guiManager, a.asset))
+	a.screen.SetNextScreen(d2gamescreen.CreateGuiTestMain(a.renderer, a.guiManager, d2util.LogLevelDefault, a.asset))
 }
```
Comment (gravestench) [inline d2app/app.go:834]: we should use the log level that is defined in the config

**Review by gravestench — CHANGES_REQUESTED** [review]
Dont use the default log level, pass the `LogLevel` in from `App.Config.LogLevel` when you create the screens.

File: d2app/app.go (hunk 1 above)
Comment (gucio321) [reply d2app/app.go]: ok @gravestanch I've corrected it.

**Review by gravestench — APPROVED** [review]

**Comment by gucio321** [comment]
I've added logger to d2game/d2player
sory for lints, I'll fix it tomorrow :-)

**Comment by gucio321** [comment]
Ok, I think, this is complete

File: d2game/d2player/globeWidget.go [hunk 2]
```diff
@@ -95,28 +88,43 @@ func newGlobeWidget(ui *d2ui.UIManager, x, y int, gtype globeType, value, valueM
...
 		value:      value,
 		valueMax:   valueMax,
 		globe:      globe,
 		overlap:    overlap,
 	}
+
+	gw.logger = d2util.NewLogger()
+	gw.logger.SetLevel(l)
+	gw.logger.SetPrefix(logPrefix)
+
```
Comment (gravestench) [inline d2game/d2player/globeWidget.go:102]: I think that this is problematic. Do individual widgets need logger instances?

should we not instead return these errors to the HUD and let the HUD report errors?

File: d2game/d2player/skillicon.go [hunk 3]
```diff
@@ -41,11 +37,24 @@ func newSkillIcon(ui *d2ui.UIManager, baseSprite *d2ui.Sprite, skill *d2hero.Her
...
 }
 
+type skillIcon struct {
+	*d2ui.BaseWidget
+	lvlLabel *d2ui.Label
+	sprite   *d2ui.Sprite
+	skill    *d2hero.HeroSkill
+
+	logger *d2util.Logger
+}
```
Comment (gravestench) [inline d2game/d2player/skillicon.go:56]: another widget that has it's own logger. maybe this is not such an issue if you assign to it an existing logger (like from the HUD or some menu)

**Review by gravestench — COMMENTED** [review]
I don't think that widgets should create loggers, they should likely use a parent logger instance (like the UI Manager logger instance). 

@essial thoughts?

File: d2game/d2player/globeWidget.go (hunk 2 above)
Comment (gucio321) [reply d2game/d2player/globeWidget.go]: yes, we can return errors to HUD, but ONLY ERRORS. What about logger.Info like in help_overlay:go:202?

File: d2game/d2player/skillicon.go (hunk 3 above)
Comment (gucio321) [reply d2game/d2player/skillicon.go:56]: maybe, so could you give some example. what should I type instead of `logger *d2util.Logger`?

**Comment by gravestench** [comment]
After a lot of discussion with @gucio321, it's not so bad if widgets have loggers. An alternate approach of using the logger of whatever was using the widget was considered, but was more trouble than it was worth. If there are no objections I would like to merge.

**Pull request merged**